packages/openapi-spec-builder/src/openapi-spec-builder.ts

```diff
@@ -132,7 +144,17 @@ export class OperationSpecBuilder {
    * @param name The name of the controller method implementing this operation.
    */
   withOperationName(name: string): this {
-    this._spec['x-operation-name'] = name;
+    return this.withExtension('x-operation-name', name);
+  }
+
+  /**
+   * Add a custom (extension) property to the operation definition.
+   *
+   * @param key The property name starting with "x-".
+   * @param value The property value.
+   */
+  withExtension(key: string, value: ExtensionValue): this {
```

Comment:
It isn't. On line 88, we are defining `OpenApiSpecBuilder.withExtension`. Here we are inside `OperationSpecBuilder` class.

Considering that almost (if not all) builders will need `withExtension` method, I'll extract a base class to contain a single shared `withExtension` implementation.